### DISCUSSION TITLE
Assigning report definition scalar values to variables in python environment

### DISCUSSION THREAD

**Discussion by rewol** [ansys/pyfluent#3350] [Q&A] [answered]
Hello,

I am trying to create a pyFluent script in which i will make derivations and calculations using the reported & computed values in the domain. For example i have an pressure inlet and i want to create a variable name mass_flow_inlet and assign mass flow rate through the inlet to my variable "mass_flow_inlet", after the solution is converged or enough iterations has been performed.

Required report definition is already defined in the case so all i need is a way to simple code snippet to get the value of this report definition and assign it to my user-defined variable, such as:

mass_flow_inlet = ....

How can i go about doing that ?

Best Regards.

**Comment by ChristianPeterKoehler** [comment]
I had a similar issue.
My "solution" was:
1. Write the report output into a file via the pyfluent command.
2. Read the file with python.
3. Extract the data value e.g. via a regex and assign it to a python variable (in your case mass_flow_inlet).
4. Delete the file again using python.
However there needs to be a better solution for this. Mine is rather a workaround and won't come in handy if you have to process a larger amount of data.

Kind Regards

Christian

**Reply by ansjschuetz** [reply, marked as answer]
Hi @rewol & all,
it may depend on the release you're working with, but when I do...
`solver.solution.report_definitions.compute(report_defs = 'report-def-0')`
...it prints comprehensive output and then _returns_ `[{'report-def-0': [0.0, 0]}]`
(...which I assume you'll be able to make further use of..[?])...

**Reply by rewol** [reply]
Thank you very much about the ideas. I ended up doing with the first suggestion.
Although i feel like there should be a simpler way for this :)

**Reply by ansjschuetz** [reply]
@rewol , did my suggestion not work? What release?
